pkg/apis/migration/v1alpha1/directvolumemigrationprogress_types.go

```diff
@@ -27,20 +27,65 @@ import (
 
 // DirectVolumeMigrationProgressSpec defines the desired state of DirectVolumeMigrationProgress
 type DirectVolumeMigrationProgressSpec struct {
-	ClusterRef *kapi.ObjectReference `json:"clusterRef,omitempty"`
-	PodRef     *kapi.ObjectReference `json:"podRef,omitempty"`
+	ClusterRef   *kapi.ObjectReference `json:"clusterRef,omitempty"`
+	PodRef       *kapi.ObjectReference `json:"podRef,omitempty"`
+	PodSelector  map[string]string     `json:"podSelector,omitempty"`
+	BackOffLimit int                   `json:"backOffLimit,omitempty"`
 }
 
+const (
+	// DVMPDoneLabelKey used to track progress of dvmp operations
+	DVMPDoneLabelKey = "migration.openshift.io/dvmp-done"
+)
+
 // DirectVolumeMigrationProgressStatus defines the observed state of DirectVolumeMigrationProgress
 type DirectVolumeMigrationProgressStatus struct {
-	Conditions                  `json:",omitempty"`
-	PodPhase                    kapi.PodPhase    `json:"phase,omitempty"`
-	ExitCode                    *int32           `json:"exitCode,omitempty"`
-	ContainerElapsedTime        *metav1.Duration `json:"containerElapsedTime,omitempty"`
-	LogMessage                  string           `json:"logMessage,omitempty"`
-	ObservedDigest              string           `json:"observedDigest,omitempty"`
-	LastObservedProgressPercent string           `json:"lastObservedProgressPercent,omitempty"`
-	LastObservedTransferRate    string           `json:"lastObservedTransferRate,omitempty"`
+	Conditions `json:",omitempty"`
+	// RsyncPodStatus observed state of most recent Rsync attempt
+	RsyncPodStatus `json:",inline"`
+	// RsyncPodStatuses history of all Rsync attempts
+	RsyncPodStatuses []RsyncPodStatus `json:"rsyncPodStatuses,omitempty"`
+	// Succeeded whether Rsync operation succeded
+	Succeeded bool `json:"succeeded,omitempty"`
+	// Failed whether Rsync operation failed
+	Failed bool `json:"failed,omitempty"`
+	// Completed wether Rsync operation completed
+	Completed bool `json:"completed,omitempty"`
```

Comment:
It looks like this field is a derived field right? you can have competed as `Failed | Succeeded` ?

Is this intended to show whether it is running, if so would it make sense to have `Running` instead of `Completed`?

Reply:
@alaypatel07 that's a good point. Earlier when I worked on API, I didn't realize that these fields were creating an unnecessary contract between DVM -> DVMP, where DVM would require DVMP to know the information. However, after discussing with you, it was clear that we should keep DVMP dumb enough and only provide "current" view of Rsync pods in the source cluster. I ended up removing these fields altogether.